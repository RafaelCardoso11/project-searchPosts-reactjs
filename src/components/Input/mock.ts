import { InputType } from "./types";
const fn = jest.fn();

export const inputPropsMock:InputType  = {
 placeHolder: 'place holder',
 type: 'text',
 value: 'value',
 event: fn
}
