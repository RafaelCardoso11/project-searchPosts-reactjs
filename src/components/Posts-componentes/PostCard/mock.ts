import { Post } from '../types';

export const postCardPropsMock: Post = {
  id: 1,
  title: 'title 1',
  body: 'body 1',
  cover: 'img/img.png',
  alt: 'title 1',
};
