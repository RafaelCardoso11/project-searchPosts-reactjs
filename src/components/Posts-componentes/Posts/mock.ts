import { Post } from "../types";

export const postsPropsMock: Post[] = [
  {
    id: 1,
    title: "title 1",
    body: "body 1",
    cover: "img/img1.png",
    alt: "title 1",
  },
  {
    id: 2,
    title: "title 2",
    body: "body 2",
    cover: "img/img2.png",
    alt: "title 2",
  },
  {
    id: 3,
    title: "title 3",
    body: "body 3",
    cover: "img/img3.png",
    alt: "title 3",
  },
];