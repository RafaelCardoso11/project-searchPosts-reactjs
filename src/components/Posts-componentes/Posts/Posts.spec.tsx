import { render, screen } from "@testing-library/react";
import { Posts } from ".";
import { postsPropsMock } from "./mock";

const props = postsPropsMock;

describe("<Posts />", () => {
  it("should render posts", () => {
    render(<Posts posts={props} />);
    expect(screen.getAllByRole("heading", { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole("img", { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
    expect(screen.getByAltText("title 3")).toHaveAttribute(
      "src",
      "img/img3.png"
    );
  });
  it("should match snapshot", () => {
    const { container } = render(<Posts posts={props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
