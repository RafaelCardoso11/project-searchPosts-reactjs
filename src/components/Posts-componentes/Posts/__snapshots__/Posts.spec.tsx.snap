// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      class="imgCard"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h1
        class="title-post"
      >
        title 1
        , 
        1
      </h1>
      <p>
        body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      class="imgCard"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h1
        class="title-post"
      >
        title 2
        , 
        2
      </h1>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      class="imgCard"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1
        class="title-post"
      >
        title 3
        , 
        3
      </h1>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
